modules/scheduler-impl/src/main/java/org/opencastproject/scheduler/endpoint/SchedulerRestService.java

```diff
@@ -1760,14 +1823,10 @@ private MediaPackage addCatalog(Workspace workspace, InputStream in, String file
     }
 
     // Update comments catalog
-    try {
-      URI catalogUrl = workspace.put(mediaPackage.getIdentifier().compact(), c.getIdentifier(), fileName, in);
-      c.setURI(catalogUrl);
-      // setting the URI to a new source so the checksum will most like be invalid
-      c.setChecksum(null);
-    } finally {
-      IOUtils.closeQuietly(in);
-    }
+    URI catalogUrl = workspace.put(mediaPackage.getIdentifier().compact(), c.getIdentifier(), fileName, in);
```

Comment:
Not familiar enough with Java, but while I would expect the garbage collector to close the InputStream `in` at some point of time, it might by preferable to do it once it is not required anymore (although the try block already misses the in.close()).

Or does workspace.put close the stream?

If you have time for a small improvement: There are two callers of this private method. Both open the InputStream without having a reference to it themselves:

`   mediaPackage = addCatalog(workspace, IOUtils.toInputStream(eventCatalog.toXmlString(), "UTF-8"),
                "dublincore.xml", MediaPackageElements.EPISODE, mediaPackage);`

It would be nicer if this method took the DublinCoreCatalog `eventCatalog` as argument. In that case, the caller could do:

`   mediaPackage = addCatalog(workspace, eventCatalog, mediaPackage);`

Idea: Open and close InputStream in same method (the arguments "dublincore.xml" and  "MediaPackageElements.EPISODE" are not relevant for the caller and could be hard-coded in this method instead).